pkg/portForward/portForward.go

```diff
@@ -75,32 +82,70 @@ func (o *PFClient) StartPortForwarding(
 		return err
 	}
 
-	portsBuf := NewPortWriter(log.GetStdout(), len(portPairsSlice), ceMapping)
+	o.originalErrorHandlers = runtime.ErrorHandlers
+	runtime.ErrorHandlers = append(runtime.ErrorHandlers, func(err error) {
+		if err.Error() == "lost connection to pod" {
+			// Stop the low-level port forwarding
+			// the infinite loop will restart it
+			if o.stopChan == nil {
+				return
+			}
+			o.stopChan <- struct{}{}
+			o.stopChan = make(chan struct{}, 1)
```

Comment:
Why are we doing this? I understand the part about `o.stopChan <- struct{}{}` is to stop the port-forwarding started using client-go library. But why are we doing `o.stopChan = make(chan struct{}, 1)` after that?

Reply:
I prefer to start with a fresh new channel. The old one will be cleaned by the GC.

Reply:
OK. But what's the rationale behind it? If it's about unblocking, as you mentioned in another comment, I think we are doing some kind of "trick".

Reply:
It's about restarting with fresh new state, because we are restarting a new port forwarding, and don't care about the state of the previous one.